cmd/internal/find/primary/numeric/predicate.go

```diff
@@ -0,0 +1,72 @@
+package numeric
+
+import (
+	"fmt"
+
+	"github.com/puppetlabs/wash/cmd/internal/find/primary/errz"
+)
+
+// Predicate is a predicate on a numeric value.
+type Predicate func(int64) bool
+
+// Parser parses numeric values.
+type Parser func(string) (int64, error)
+
+// ParsePredicate parses a Predicate from str. Str should
+// satisfy the regex `(\+|\-)?<number>`, where <number> is s.t.
+// that parser(<number>) does not return an error for at least one
+// parser in parsers. The returned value is the parsed predicate
+// and the id of the parser that parsed <number>.
+func ParsePredicate(str string, parsers ...Parser) (Predicate, int, error) {
+	if len(str) == 0 {
+		return nil, -1, errz.NewMatchError("empty input")
+	}
+	if len(parsers) == 0 {
+		panic("numeric.ParsePredicate called without any parsers")
+	}
+
+	cmp := str[0]
+	if cmp == '+' || cmp == '-' {
+		str = str[1:]
+	} else {
+		cmp = '='
+	}
+
+	var parserID int
+	var n int64
+	var err error
+	for i, parser := range parsers {
```

Comment:
This logic's similar to the `try` function in `meta/grammar.go`. Unfortunately, I can't abstract that logic out b/c I don't want to do unnecessary type assertions in numeric predicates.